sdk/include/system/DateTime.hpp

```diff
@@ -282,7 +295,27 @@ class DateTime final
     *
     * @return Success if in_timeStr is a valid ISO 8601 representation of a date and time; Error otherwise.
     */
-   static Error fromString(const std::string& in_timeStr, DateTime& out_dateTime);
+    
+    template <typename TimeType>
+     static std::string format(const TimeType& time,
+                   const std::string& format)
+      {
+      using namespace boost::posix_time;
+
+      // facet for http date (construct w/ a_ref == 1 so we manage memory)
+      time_facet httpDateFacet(1);
+      httpDateFacet.format(format.c_str());
+
+       // output and return the date
+         std::ostringstream dateStream;
+         dateStream.imbue(std::locale(dateStream.getloc(), &httpDateFacet));
+         dateStream << time;
+         return dateStream.str();
+      }
+
+static Error fromString(const std::string& in_timeStr,DateTime& out_dateTime);
+static Error fromString(const std::string& in_timeStr, const std::string& in_format, DateTime& out_dateTime);
```

Comment:
These are missing comments and the whitespace is off. The comment for the first function is above the format function currently.